### PR TITLE
Refactor/term ui

### PR DESCRIPTION
**Pull request by tomchentw**
## Depends on

#19 

## Description

* Add `prettier` with `lint-saged` & `husky` support to `src/components/` folder
* Add `classnames` to dependencies
* Extract several `Row` related components with three type of builders

### CODE REVIEW

**Comment by robertabcd** [comment]
React's official site seems to have ditched jsx and just use js.
https://reactjs.org/docs/faq-structure.html
Probably you can take this chance to do that, as you are moving files.

**Comment by tomchentw** [comment]
@robertabcd sure thing. Will use `.js` from now on.

---

Update: Migrate to `.js` in the last two commits.

**Comment by tomchentw** [comment]
Done rebasing.

**Comment by tomchentw** [comment]
Ping @robertabcd for reviewing

File: src/components/Row/LinkSegmentBuilder.jsx [hunk 1]
```diff
@@ -0,0 +1,82 @@
+import cx from "classnames";
+import { HyperLinkPreview } from "../../js/image_preview";
+import HyperLink from "./HyperLink";
+import ColorSegmentBuilder from "./ColorSegmentBuilder";
+
+export class LinkSegmentBuilder {
+  constructor(row, showsLinkPreviews, forceWidth, highlighted) {
+    this.row = row;
+    this.forceWidth = forceWidth;
+    this.highlighted;
```
Comment (robertabcd) [inline src/components/Row/LinkSegmentBuilder.jsx]: = highlighted?

File: src/components/Row/LinkSegmentBuilder.jsx [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+    //
+    this.segs = [];
+    this.linkPreviews = showsLinkPreviews ? [] : false;
+    //
+    this.colorSegBuilder = null;
+    this.col = null;
+    this.href = null;
+  }
+
+  beginSegment() {
```
Comment (robertabcd) [inline src/components/Row/LinkSegmentBuilder.jsx]: This pushes a segment into the array. Maybe name it `saveSegment` or something?

**Review by robertabcd — COMMENTED** [review]
Sorry for the delay. Please see the comments inline. I really appreciate your work on modernizing it. :)

File: src/components/Row/LinkSegmentBuilder.jsx (hunk 1 above)
Comment (tomchentw) [reply src/components/Row/LinkSegmentBuilder.jsx]: Thanks for catching this!

File: src/components/Row/LinkSegmentBuilder.jsx (hunk 2 above)
Comment (tomchentw) [reply src/components/Row/LinkSegmentBuilder.jsx]: Good suggestion. Renamed

**Comment by tomchentw** [comment]
Thank for reviewing, @robertabcd.

Please feel free to use `Request Changes` when reviewing in such situations. All issues are fixed now.

**Pull request merged**